### PR TITLE
helprank/data -- wire-up userBookmarkClicks table

### PR DESCRIPTION
**Pull request by raymondng**
- rekeep counts 

@yingjieMiao @ymatsuda

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/AttributionCommander.scala [hunk 1]
```diff
@@ -73,4 +80,45 @@ class AttributionCommander @Inject() (
     usersByDeg zip rekeepsByDeg
   }
```
Comment (yingjieMiao) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/AttributionCommander.scala:82]: our convention is to have a space after `:`

Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/AttributionCommander.scala:82]: Didn't know that. Thanks -- will take care of it.
We probably need a lint tool

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/AttributionCommander.scala [hunk 2]
```diff
@@ -73,4 +80,45 @@ class AttributionCommander @Inject() (
...
 
+  def updateReKeepStats(userId:Id[User], n:Int = 3): Future[Seq[UserBookmarkClicks]] = { // expensive -- admin only
+    val rekeepCountsF = db.readOnlyAsync(dbMasterSlave = Slave) { implicit ro =>
+      rekeepRepo.getReKeepsByKeeper(userId).groupBy(_.keepId).map { case (keepId, rekeeps) =>
+        (keepId, rekeeps.head.uriId) -> rekeeps.length
+      }
+    }
+    rekeepCountsF map { rekeepCounts =>
+      val rekeepStats = rekeepCounts.map { case ((keepId, uriId), rekeepCount) =>
+        (keepId, uriId) -> (rekeepCount, getReKeepsByDegree(userId, keepId, n).map(_._1).flatten.length - 1)
```
Comment (yingjieMiao) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/AttributionCommander.scala]: as we discussed, the `-1` is a bit confusing. One need to guess or read the repo code to know we are excluding the keep itself. So, we probably can add some comments in the repo. Or, having two different endpoints, or use a boolean Flag. Just to reduce some surprise.

Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/AttributionCommander.scala]: Good point. I'll add a comment to clarify it.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/AttributionCommander.scala [hunk 3]
```diff
@@ -73,4 +80,45 @@ class AttributionCommander @Inject() (
...
+      }
+      log.info(s"[updateReKeepStats($userId)] rekeepStats=${rekeepStats.mkString(",")}")
+
+      val res = db.readWrite { implicit rw =>
+        rekeepStats.map{ case ((keepId, uriId), (rkCount, aggCount)) =>
+          userBookmarkClicksRepo.getByUserUri(userId, uriId) match {
+            case None =>
+              log.error(s"[updateReKeepStats($userId)] couldn't find bookmarkClick entry for $uriId")
```
Comment (yingjieMiao) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/AttributionCommander.scala:99]: this is based on the assumption that bookmark click is the only source of rekeeping.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/AttributionCommander.scala (hunk 1 above)
Comment (yingjieMiao) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/AttributionCommander.scala:82]: thanks

**Comment by yingjieMiao** [comment]
lgtm

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/AttributionCommander.scala (hunk 3 above)
Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/AttributionCommander.scala:99]: for now, yes -- but this will change going forward.

**Pull request merged**